src/Uno.SourceGeneration.Engine.Shared/InternalGeneratorExecutionContext.cs

```diff
@@ -45,6 +46,7 @@ public override void AddSource(string hintName, SourceText sourceText)
 
 		public override void ReportDiagnostic(Diagnostic diagnostic)
 		{
+			throw new InvalidOperationException(diagnostic.GetMessage());
```

Comment:
We probably should not throw, as it would change the behavior at the call site. Maybe we could keep the diagnostics for the end of the generation and raise everything at once, not too sure here.